service/src/main/java/tds/exam/services/impl/ExamServiceImpl.java

```
@@ -273,6 +276,86 @@ public ExamServiceImpl(ExamQueryRepository examQueryRepository,
         return Optional.empty();
     }
 
+    @Override
+    public Response<ExamConfiguration> startExam(final UUID examId) {
+        ExamConfiguration examConfig = null;
+
+        try {
+            Exam exam = examQueryRepository.getExamById(examId)
+                .orElseThrow(() -> new IllegalArgumentException(String.format("No exam found for id %s", examId)));
```

I feel this should throw a NotFound exception.